chia/full_node/weight_proof.py

```diff
@@ -1160,9 +1167,11 @@ def sub_slot_data_vdf_input(
                     if ssd.cc_slot_end_info is not None:
                         ssd = sub_slots[ssd_idx + 1]
                         break
+                    assert ssd.total_iters is not None
                     if not (ssd.total_iters > sp_total_iters):
```

Comment:
all of these comparisons with `ssd.total_iters` seem to assume `ssd.total_iters` is never `None`, even though it can be.
I'm unsure whether this is a reasonable assumption, or whether the logic should be changed here to handle this case in some special way.
adding the asserts here makes the current assumptions explicit.

Reply:
Maybe `SubSlotData` could be generic against this?  It's frozen, so it's either `None` or not forever.

Reply:
what do you mean?
these asserts only hold in the specific places where they are placed, without the surrounding checks  we cannot make the assumption that `total_iters` is not `None`